packages/webdriver-utils/src/providers/genericProvider.js

```diff
@@ -92,4 +101,104 @@ export default class GenericProvider {
   async setDebugUrl() {
     this.debugUrl = 'https://localhost/v1';
   }
+
+  async findIgnoredRegions(ignoreRegionXpaths, ignoreRegionSelectors, ignoreRegionSeleniumElements, customIgnoreRegions) {
+    const ignoredElementsArray = [];
+    await this.ignoreRegionsByXpaths(ignoredElementsArray, ignoreRegionXpaths);
+    await this.ignoreRegionsBySelector(ignoredElementsArray, ignoreRegionSelectors);
+    await this.ignoreRegionsByElement(ignoredElementsArray, ignoreRegionSeleniumElements);
+    await this.addCustomIgnoreRegions(ignoredElementsArray, customIgnoreRegions);
+
+    const ignoredElementsLocations = {
+      ignoreElementsData: ignoredElementsArray
+    };
+
+    return ignoredElementsLocations;
+  }
+
+  async ignoreElementObject(selector, elementId) {
+    const scaleFactor = await this.metaData.devicePixelRatio();
+    const rect = await this.driver.rect(elementId);
+    const location = { x: parseInt(rect.x), y: parseInt(rect.y) };
+    const size = { height: parseInt(rect.height), width: parseInt(rect.width) };
+    const coOrdinates = {
+      top: location.y * scaleFactor,
+      bottom: (location.y + size.height) * scaleFactor,
+      left: location.x * scaleFactor,
+      right: (location.x + size.width) * scaleFactor
+    };
+
+    const jsonObject = {
+      selector,
+      coOrdinates
+    };
+
+    return jsonObject;
+  }
+
+  async ignoreRegionsByXpaths(ignoredElementsArray, xpaths) {
+    for (const xpath of xpaths) {
+      try {
+        const element = await this.driver.findElement('xpath', xpath);
+        const selector = `xpath: ${xpath}`;
+        const ignoredRegion = await this.ignoreElementObject(selector, element.ELEMENT);
+        ignoredElementsArray.push(ignoredRegion);
+      } catch (e) {
+        log.warn(`Selenium Element with xpath: ${xpath} not found. Ignoring this xpath.`);
+        log.debug(e.toString());
+      }
+    }
+  }
+
+  async ignoreRegionsBySelector(ignoredElementsArray, ids) {
+    for (const id of ids) {
+      try {
+        const element = await this.driver.findElement('css selector', id);
+        const selector = `id: ${id}`;
+        const ignoredRegion = await this.ignoreElementObject(selector, element.ELEMENT);
+        ignoredElementsArray.push(ignoredRegion);
+      } catch (e) {
+        log.warn(`Selenium Element with id: ${id} not found. Ignoring this id.`);
+        log.debug(e.toString());
+      }
+    }
+  }
+
+  async ignoreRegionsByElement(ignoredElementsArray, elements) {
+    for (let index = 0; index < elements.length; index++) {
+      try {
+        const selector = `element: ${index}`;
+
+        const ignoredRegion = await this.ignoreElementObject(selector, elements[index]);
+        ignoredElementsArray.push(ignoredRegion);
+      } catch (e) {
+        log.warn(`Correct Web Element not passed at index ${index}.`);
+        log.debug(e.toString());
+      }
+    }
+  }
```

Comment:
We should avoid mutations in the array. Lets do so that the functions return an array and we concat all in the end.

Also, all this can be replaced by a single `ignoreRegionsBy(findElementFn, elements)`